client/state/document-head/reducer.js

```diff
@@ -23,7 +23,6 @@ export const title = createReducer(
 	'',
 	{
 		[ DOCUMENT_HEAD_TITLE_SET ]: ( state, action ) => action.title,
-		[ ROUTE_SET ]: () => '',
```

Comment:
Principles of symmetry seem to suggest that we should get rid of the respective `[ ROUTE_SET ]` resets in the two other reducers in this file, too...

Reply:
Hmm, OTOH, seems like that's what #8224 was about in the first place...

Reply:
All that seems to indicate that our `document-head` Redux state and component still aren't quite there yet. This is exacerbated by the fact that in some instances, we're manually dispatching the corresponding Redux actions inside section controllers (rather than using the `<DocumentHead />` componet -- my bad, that was part of a semi-automated refactor)...

Reply:
The `ROUTE_SET` in the other reducers is essential to reset these things to the initial state:

- The unread count, if any (e.g. `(10) Foo < Bar - WordPress.com`), only modified by the [Reader](https://github.com/Automattic/wp-calypso/blob/master/client/reader/update-notice/index.jsx#L41).
- The `<head>` links and meta tags, only modified by the [login](https://github.com/Automattic/wp-calypso/blob/master/client/login/wp-login/index.jsx#L174-L178), [themes](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/themes/theme-showcase.jsx#L221), and [single theme](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/theme/main.jsx#L600) pages.

As far as I can see, we need them to stay, because we indeed need them `unreadCount`, `link`, and `meta` to reset to their initial state when we're not in one of the few routes that use them.

Reply:
See https://github.com/Automattic/wp-calypso/pull/23961#issuecomment-383975586 for evidence...